src/timeMachine/components/view_options/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {FC, useState} from 'react'
+import {useSelector} from 'react-redux'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+
+import {getVisTable} from 'src/timeMachine/selectors'
+
+import {useVisXDomainSettings} from 'src/shared/utils/useVisDomainSettings'
+
+interface Props {
+  setDomain: (xDomain: [number, number]) => void
+  xDomain: number[] | null
```

Comment:
@TCL735 is this `number [] | null` okay for `xDomain`? i've only seen `xDomain` typed to `number[]` in our code. the documentation in giraffe says it should be a number array or it shouldn't exist, which is `undefined`